### ISSUE TITLE
CA1810 should not warn when not assigning any fields

### ISSUE THREAD

**Issue by JohanLarsson** [dotnet/roslyn-analyzers#4409]
```cs
public static class Drag
{
    static Drag()
    {
        EventManager.RegisterClassHandler(typeof(UIElement), UIElement.MouseLeftButtonDownEvent, new MouseButtonEventHandler(OnMouseLeftButtonDown));

        static void OnMouseLeftButtonDown(object sender, MouseEventArgs e)
        {
            var element = sender as UIElement;
            if (element?.GetWithMouse() != true)
            {
                return;
            }

            DraggedItem.SetTarget(element);
            var window = Window.GetWindow(element);
            mousePreviousPosition = e.GetPosition(window);
            elementPosition = new Point(Canvas.GetLeft(element), Canvas.GetTop(element));
            _ = element.CaptureMouse();
        }
    }
}
```

**Comment by Evangelink** [comment]
I think the example has been cut too much, it seems that `mousePreviousPosition` and `elementPosition` are missing, are they fields? local variable?

Also which version of the analyzer are you using?

**Comment by Evangelink** [comment]
@JohanLarsson Shall I assume it was using fields?

**Comment by JohanLarsson** [comment]
It should not warn if using fields in callback like that but you were right the code is too trimmed.

**Comment by Evangelink** [comment]
I think you are right that it should not report, we have been doing a few changes to support this so I will keep your ticket open and check if it is still reporting on master.

**Comment by JohanLarsson** [comment]
I found a place I had disabled the warning via pragma and there was no warning so looks like things have changed for the better. I'll provide better repro code if I see there warning where it should not warn again.
